server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -334,4 +360,116 @@ private CircuitBreaker validateAndCreateBreaker(BreakerSettings breakerSettings)
                 this,
                 breakerSettings.getName());
     }
+
+    private static OverLimitStrategy createDoubleCheckStrategy(JvmInfo jvmInfo, LongSupplier currentMemoryUsageSupplier,
+                                                               LongSupplier timeSupplier, long minimumInterval) {
+        if (jvmInfo.useG1GC().equals("true")
```

Comment:
In line with an earlier comment, we could pass in `trackRealMemoryUsage` to this method and add it to this check.

Reply:
Also part of 3f61f93f6660b32153047b7fb0393b314510d238